src/middlewared/middlewared/etc_files/initiators.allow.mako

```diff
@@ -1,7 +1,18 @@
 <%
+    import ipaddress
     base_name = middleware.call_sync('iscsi.global.config')['basename']
     targets = middleware.call_sync('iscsi.target.query', [['auth_networks', '!=', []]])
+
+    def parse_auth(auth):
+        try:
+            ipobj = ipaddress.ip_interface(auth)
+        except ValueError:
+            middleware.logger.warning("Invalid IP address: %s", auth, exc_info=True)
+        else:
+            if ipobj.network.prefixlen in (32, 128):
+                return str(ipobj.ip)
+            return str(ipobj.network)
 %>\
 % for target in targets:
-${base_name}:${target['name']} ${', '.join(target['auth_networks'])}
+${base_name}:${target['name']} ${', '.join([parse_auth(auth) for auth in target['auth_networks']])}
```

Comment:
Sorry, I just noticed this possible failure condition. If `parse_auth` returns `None` then `.join` will fail with `TypeError`

Reply:
Think it might be best to just pass `target['auth_networks']` in it's entirety to `parse_auth` and make `parse_auth` return the formatted string.

Reply:
Sounds good.  See below.